packages/generator/src/service-generator.ts

```diff
@@ -0,0 +1,196 @@
+      serviceMetadata,
+      serviceDefinitionPaths
+    );
+    const vdmServiceEntities = this.getServiceEntities(
+      serviceMetadata,
+      serviceDefinitionPaths
+    );
+
+    return {
+      ...vdmServicePackageMetaData,
+      ...vdmServiceEntities
+    };
+  }
+
+  private getServicePackageMetaData(
+    serviceMetadata: ServiceMetadata,
+    serviceDefinitionPaths: ServiceDefinitionPaths
+  ): VdmServicePackageMetaData {
+    const directoryName = this.globalNameFormatter.uniqueDirectoryName(
+      ServiceNameFormatter.originalToServiceName(
+        serviceMetadata.edmx.namespace
+      ),
+      serviceMetadata.edmx.fileName
+    );
+    const npmPackageName = this.globalNameFormatter.uniqueNpmPackageName(
+      npmCompliantName(directoryName),
+      serviceMetadata.edmx.fileName
+    );
+    const speakingModuleName = ServiceNameFormatter.directoryToSpeakingModuleName(
+      directoryName
+    );
+    const className = `${speakingModuleName.replace(/ /g, '')}`;
+
+    return {
+      oDataVersion: serviceMetadata.edmx.oDataVersion,
+      namespace: serviceMetadata.edmx.namespace,
+      originalFileName: serviceMetadata.edmx.fileName,
+      directoryName,
+      npmPackageName,
+      speakingModuleName,
+      servicePath: getServicePath(
+        serviceMetadata,
+        this.serviceMapping[serviceMetadata.edmx.fileName]
+      ),
+      edmxPath: serviceDefinitionPaths.edmxPath,
+      apiBusinessHubMetadata: apiBusinessHubMetadata(serviceMetadata.swagger),
+      className
+    };
+  }
+
+  private getServiceEntities(
+    serviceMetadata: ServiceMetadata,
+    serviceDefinitionPaths: ServiceDefinitionPaths
+  ): VdmServiceEntities {
+    const formatter = new ServiceNameFormatter();
+
+    /*
+    Do function imports before complex types so that function with name "createSomething" gets a nicer name, because the builder function of a complex type would also be called `create${ComplexTypeName}`.
+    */
+    const functionImportsWithoutReturnType = isV2Metadata(serviceMetadata.edmx)
+      ? generateFunctionImportsV2(serviceMetadata, formatter)
+      : generateFunctionImportsV4(serviceMetadata, formatter);
+
+    const complexTypes = isV2Metadata(serviceMetadata.edmx)
+      ? generateComplexTypesV2(serviceMetadata, formatter)
+      : generateComplexTypesV4(serviceMetadata, formatter);
+
+    const entities = isV2Metadata(serviceMetadata.edmx)
+      ? generateEntitiesV2(serviceMetadata, complexTypes, formatter)
+      : generateEntitiesV4(serviceMetadata, complexTypes, formatter);
+
+    const functionImports = parseReturnTypes(
+      functionImportsWithoutReturnType,
+      entities,
+      complexTypes
+    );
+
+    return {
+      complexTypes,
+      entities,
+      functionImports
+    };
+  }
+
+  private readEdmxAndSwaggerFile(
+    serviceDefinitionPaths: ServiceDefinitionPaths
+  ): ServiceMetadata {
+    const serviceMetadata: ServiceMetadata = {
+      // TODO: pass parameter
```

Comment:
TODO?

Reply:
hm, to old to bad written. Forgott already what I wanted to do. Note to myself. Either make a better TODO or do it right away.